kratos/geometries/geometry.h

```diff
@@ -3176,20 +3182,45 @@ class Geometry
 
 
 private:
-    ///@name Static Member Variables
-    ///@{
-
-    // static const GeometryData msEmptyGeometryData;
-
-    ///@}
     ///@name Member Variables
     ///@{
 
+    std::uint64_t mId;
+
     GeometryData const* mpGeometryData;
 
     static const GeometryDimension msGeometryDimension;
 
     PointsArrayType mPoints;
+
+    ///@}
+    ///@name Id Bit-Change Operations
+    ///@{
+
+    /// Checks first bit in Id. 0 -> id; 1 -> name/ string
+    static inline bool IsIdGeneratedFromString(IndexType Id)
+    {
+        return Id & (IndexType(1) << 63);
+    }
+
+    /// Sets first bit in Id to 1 -> name/ string
+    static inline void SetIdGeneratedFromString(IndexType& Id)
+    {
+        Id |= (IndexType(1) << 63);
+    }
+
+    /// Checks second bit in Id. 0 -> defined id; 1 -> self assigned
+    static inline bool IsIdSelfAssigned(IndexType Id)
+    {
+        return Id & (IndexType(1) << 62);
```

Comment:
And here to` return Id & (IndexType(1) << (sizeof(IndexType) - 2)); `to be compatible with 32 bit

Reply:
is resolved.